ush/remap.sh

```diff
@@ -0,0 +1,376 @@
+#   by splitting the input string at commas and then removing leading
+#   and trailing spaces.
+#
+# Parameters:
+#   $1 - The comma-delimited string to split.
+#
+# Global Variables:
+#   global_array - An array containing the split elements.
+#
+# Example usage:
+#   _comma_split_string "item1,item2 item3,item4"
+#   for element in "${global_array[@]}"; do
+#       echo "$element"
+#   done
+#
+# This example will split the input string into individual elements
+# and print each element on a separate line.
+function _comma_split_string() {
+    local string="${1}"
+
+    local local_array=()
+    global_array=()
+    IFS="," read -ra items <<< "${string}"
+    for item in "${items[@]}"; do
+        local_array+=("${item} ")
+    done
+    for item in "${local_array[@]}"; do
+        IFS=" " read -ra items <<< "${item}"
+        for element in "${items[@]}"; do
+            global_array+=("${element} ")
+        done
+    done
+}
+
+#######
+
+# _strip_whitespace - Remove whitespace from a string.
+#
+# Description:
+#   This function takes an input string and removes all whitespace
+#   characters (spaces, tabs, and newline characters) to produce a
+#   cleaned output string.
+#
+# Parameters:
+#   $1 - The input string from which whitespace will be removed.
+#
+# Return:
+#   The cleaned string with no whitespace.
+#
+# Example usage:
+#   cleaned_string=$(_strip_whitespace "   This is a string   with spaces  ")
+#   echo "Cleaned string: \"$cleaned_string\""
+#
+# This example will remove all leading, trailing, and internal
+# whitespace from the input string and display the cleaned result.
+function _strip_whitespace(){
+    local in_string="${1}"
+
+    out_string=$(echo "${in_string}" | sed "s/ //g")
+}
+
+#######
+
+# nc_concat - Concatenate netCDF files and manage file paths.
+#
+# Description:
+#   This function performs the concatenation of two netCDF files:
+#   `output_path` and `var_interp_path`. If the `output_path` file
+#   exists, the function merges `var_interp_path` into it and manages
+#   temporary files accordingly. If `output_path` does not exist, it
+#   creates a new netCDF file by renaming `var_interp_path` to
+#   `output_path`.
+#
+# Parameters:
+#   $1 - The path to the variable-interpolated netCDF file.
+#
+# Global Variables:
+#   output_path - The path to the output netCDF file.
+#
+# Example usage:
+#   nc_concat "variable_interpolated.nc"
+#
+# This example will concatenate the variable-interpolated file into
+# the output netCDF file, handling the case where the output file may
+# or may not exist.
+function nc_concat(){
+    local var_interp_path="${1}"
+
+    tmp_nc_file="${PWD}/tmp_nc.nc"
+    if test -e "${output_path}"; then
+	echo "netCDF-formatted file path ${output_path} exists; merging ${var_interp_path}"
+	cdo merge "${output_path}" "${var_interp_path}" "${tmp_nc_file}"
+	mv "${tmp_nc_file}" "${output_path}"
+	rm "${var_interp_path}" >> /dev/null
```

Comment:
How damaging is it if the file isn't removed, and why is STDOUT being trashed?

Reply:
This is to remove files that are no longer needed. Once the interpolated file, for the respective variable, is merged (i.e., appended) to the output netCDF file path it is no longer needed. If the files remain they will contribute to the disk footprint without reason.

Reply:
Well, what I was getting at was: should this be `rm -f` to make sure it gets removed? And still not sure why the STDOUT is being redirected to `/dev/null`.

Reply:
I piped the stdout/stderr to a file in in the runtime directory. See `REMAP_LOG`.

Reply:
Still not sure why this is even necessary. It's not like `rm` prints out a bunch of garbage output.